pkg/endpoint/bpf.go

```diff
@@ -563,10 +563,14 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir string, regenContext *Regene
 		closeChan := loadinfo.LogPeriodicSystemLoad(log.WithFields(logrus.Fields{logfields.EndpointID: epID}).Debugf, time.Second)
 
 		// Compile and install BPF programs for this endpoint
-		stats.bpfCompilation.Start()
 		ctx, cancel := context.WithTimeout(context.Background(), ExecTimeout)
-		err = loader.CompileAndLoad(ctx, epInfoCache)
-		stats.bpfCompilation.End()
+		if bpfHeaderfilesChanged {
+			stats.bpfCompilation.Start()
+			err = loader.CompileAndLoad(ctx, epInfoCache)
+			stats.bpfCompilation.End()
+		} else {
+			err = loader.ReloadDatapath(ctx, epInfoCache)
+		}
```

Comment:
Would be great to make the log message below specific to compilation or reload so we know whether a reload or a recompilation was performed.

Reply:
Would this be clear enough?

```
diff --git a/pkg/endpoint/bpf.go b/pkg/endpoint/bpf.go
index c2b4eba1918c..a4cb016e697a 100644
--- a/pkg/endpoint/bpf.go
+++ b/pkg/endpoint/bpf.go
@@ -562,12 +562,16 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir string, regenContext *Regene
        if bpfHeaderfilesChanged || regenContext.ReloadDatapath {
                closeChan := loadinfo.LogPeriodicSystemLoad(log.WithFields(logrus.Fields{logfields.EndpointID: epID}).Debugf, time.Second)

+               scopedLog := e.Logger()
+
                // Compile and install BPF programs for this endpoint
+               compilationTime := "0"
                ctx, cancel := context.WithTimeout(context.Background(), ExecTimeout)
                if bpfHeaderfilesChanged {
                        stats.bpfCompilation.Start()
                        err = loader.CompileAndLoad(ctx, epInfoCache)
                        stats.bpfCompilation.End()
+                       compilationTime = stats.bpfCompilation.Total().String()
                } else {
                        err = loader.ReloadDatapath(ctx, epInfoCache)
                }
@@ -575,8 +579,8 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir string, regenContext *Regene
                close(closeChan)

                e.Logger().WithError(err).
-                       WithField(logfields.BPFCompilationTime, stats.bpfCompilation.Total().String()).
-                       Info("Recompiled endpoint BPF program")
+                       WithField(logfields.BPFCompilationTime, compilationTime).
+                       Info("Reloaded endpoint BPF program")

                if err != nil {
                        return epInfoCache.revision, compilationExecuted, err
```